src/devices_models/devices/storage.jl

```diff
@@ -335,64 +347,66 @@ end
-) where {St <: PSY.BatterywEMS, S <: PM.AbstractPowerModel}
+) where {St <: PSY.BatteryEMS, S <: PM.AbstractPowerModel}
 
     parameters = model_has_parameters(psi_container)
     use_forecast_data = model_uses_forecasts(psi_container)
-
-    target_forecast_label = "storage_target"
-    constraint_infos_target = Vector{DeviceTimeSeriesConstraintInfo}(undef, length(devices))
-    for (ix, d) in enumerate(devices)
-        ts_vector_target = get_time_series(psi_container, d, target_forecast_label)
-        constraint_info_target = DeviceTimeSeriesConstraintInfo(
-            d,
-            x -> PSY.get_storage_target(x) * PSY.get_storage_capacity(x),
-            ts_vector_target,
-        )
-        constraint_infos_target[ix] = constraint_info_target
-    end
-
-    if parameters
-        energy_soft_target_timeseries_param(
-            psi_container,
-            constraint_infos_target,
-            make_constraint_name(ENERGY_TARGET, St),
-            (
-                make_variable_name(ENERGY, St),
-                make_variable_name(ENERGY_TARGET_SLACK, St),
-            ),
-            UpdateRef{St}(TARGET, target_forecast_label),
-        )
+    if !use_forecast_data
+        error("EndOfPeriodEnergySoftTarget formulation is supported only with forecast data, please consider using EndOfPeriodEnergyStaticSoftTarget for $St")
     else
-        energy_soft_target_timeseries(
-            psi_container,
-            constraint_infos_target,
-            make_constraint_name(ENERGY_TARGET, St),
-            (
-                make_variable_name(ENERGY, St),
-                make_variable_name(ENERGY_TARGET_SLACK, St),
-            ),
-        )
+        target_forecast_label = "storage_target"
+        constraint_infos_target = Vector{DeviceTimeSeriesConstraintInfo}(undef, length(devices))
+        for (ix, d) in enumerate(devices)
+            ts_vector_target = get_time_series(psi_container, d, target_forecast_label)
+            constraint_info_target = DeviceTimeSeriesConstraintInfo(
+                d,
+                x -> PSY.get_rating(x),
+                ts_vector_target,
+            )
+            constraint_infos_target[ix] = constraint_info_target
+        end
+        if parameters
+            energy_soft_target_timeseries_param(
+                psi_container,
+                constraint_infos_target,
+                make_constraint_name(ENERGY_TARGET, St),
+                (
+                    make_variable_name(ENERGY, St),
+                    make_variable_name(ENERGY_TARGET_SLACK, St),
+                ),
+                UpdateRef{St}(TARGET, target_forecast_label),
+            )
+        else
+            energy_soft_target_timeseries(
+                psi_container,
+                constraint_infos_target,
+                make_constraint_name(ENERGY_TARGET, St),
+                (
+                    make_variable_name(ENERGY, St),
+                    make_variable_name(ENERGY_TARGET_SLACK, St),
```

Comment:
I had previously recognized a pattern that all `*._SLACK` variables were subject to a `*._SLACK_COST` defined in https://github.com/NREL-SIIP/PowerSimulations.jl/blob/sd/battery_model/src/core/definitions.jl. This breaks that convention and makes this slack subject to the cost defined in the struct. 
I'm not necessarily saying that we should keep the old convention and change this terminology, but rather highlighting the change and asking if it matters to anyone (@jd-lara)?

Reply:
`ENERGY_TARGET_SHORTAGE`/`ENERGY_SHORTAGE`?